app/controllers/application_controller.rb

```diff
@@ -22,6 +22,26 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
+  #By default only show breadcrumbs to logged-in users.
+  def show_breadcrumbs
+    current_ability.current_user.logged_in?
+  end
+  helper_method :show_breadcrumbs
+
+  # This is used in generic_works_controller and collection_show_controller.
+  # It resolves issue https://github.com/sciencehistory/chf-sufia/issues/885
+  def special_breadcrumb_filter (the_breadcrumbs)
+    return if the_breadcrumbs == nil
+    # If the user's logged in, leave as is.
+    return if current_ability.current_user.logged_in?
+    # Otherwise, only show the "Back to Search Results" breadcrumb,
```

Comment:
should be using the `show_breadcrumbs?` method you defined above, as `current_ability.current_user.logged_in?` Or do you mean to actually check logged in here, regardless of whether the controller has redefined `show_breadcrumbs?`? 

I don't entirely understand what this special filter is doing -- why isn't the `show_breadcrumbs?`method enough, why do you need to special filter even when you're showing breadcrumbs?

In general, this PR looks like a pretty sweet solution though, nice job!

Reply:
Maybe just call the method `filter_breadcrumbs`?

Also standard ruby style is _no_ space in between method name and param paranthesis, in both method definitions and method calls.